### PR TITLE
New user sequence

### PR DESCRIPTION
**Pull request by pbywater**
Tutorial style intro when user has no memories

Relates to #108 #83

### CODE REVIEW

File: src/client/d3/newUserIntro.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+              .style('opacity', 1);
+      d3.select('.text-group-3').on('click', () => {
+        d3.select('.text-group-3')
+                  .attr('class', 'text-group-4');
+        d3.select('.tap')
+            .transition()
+              .duration(500)
+              .style('opacity', 0);
+        d3.select('.welcome')
+                    .transition()
```
Comment (des-des) [inline src/client/d3/newUserIntro.js]: wtf indentation

File: src/client/d3/newUserIntro.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  });
+
+
+  const welcome = textGroup
+          .append('text')
+          .attr('class', 'welcome')
+          .text('Welcome to Echo')
+          .attr('fill', 'white')
+          .style('opacity', 0)
+                      .transition()
```
Comment (des-des) [inline src/client/d3/newUserIntro.js]: This indentation is ridiculous

File: src/client/d3/newUserIntro.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  .attr('class', 'memory-group')
+  .call(d3.zoom()
+    .scaleExtent([1 / 3, 3])
+    .on('zoom', zoomed));
+  const nodeGrp = fdGrp
+      .append('g')
+        .attr('class', 'nodes');
+
+  const nodes = nodeGrp
+      .selectAll('circle.node')
```
Comment (des-des) [inline src/client/d3/newUserIntro.js]: There is too much repetition here,

You are adding lots of complexity + branching with this feature.

Is there no way you can use the old functionality? Just pass in you mock data and run the tutorial, You are redefining to much stuff here.

Its not maintainable or good

**Review by des-des — CHANGES_REQUESTED** [review]
I am really not sure about this.

First, you need to fix the indentation.

There is too much repetition, you are repeating lots of functionality.

**Comment by pbywater** [comment]
@des-des I've fixed the horrible indentation (sorry!!!) and taken out as much code as I can. Weirdly we can't get the drag functions of similar to work if they move outside of the d3 file that they're being used in. The code is very repetitive with the timeouts and text changes but not sure how else to do this?

**Comment by des-des** [comment]
As discussed yesterday, going to merge with master and try and run the dummy data though the old functionality

**Review by des-des — APPROVED** [review]

**Pull request merged**